src/controllers/TalksController.php

```diff
@@ -123,37 +133,307 @@ public function postAction($request, $db) {
+            FILTER_SANITIZE_URL
+        );
+
+        $new_id = $talk_mapper->createTalk($talk);
+        $event_mapper->cacheTalkCount($talk['event_id']);
+
+        $incoming_speakers_list = (array) $request->getParameter('speakers');
+        $speakers = array_map(function($speaker){
+            $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
+            $speaker = trim($speaker);
+            return $speaker;
+        }, $incoming_speakers_list);
+
+        foreach($speakers as $speaker) {
+            $talk['speakers'][] = filter_var($speaker, FILTER_SANITIZE_STRING);
+        }
+
+        $uri = $request->base . '/' . $request->version . '/talks/' . $new_id;
+        header("Location: " . $uri, true, 201);
+
+        $new_talk = $talk_mapper->getTalkById($new_id);
+        return $new_talk;
+    }
+
+    /**
+     * Edit a talk
+     *
+     * This action expects the following parameters:
+     *
+     * - string title
+     * - string description
+     * - string language (has to be a value from the languages-table
+     * - string slides_link (has to be a URL)
+     * - string type (has to be a value from the categories-table
+     * - int duration
+     * - string date (will be parsed as datetime - include start time as well!)
+     *
+     * Users with admin-rights on the talk (the admins as well as the event-hosts)
+     * can edit all fields, verified speakers will only be able to edit title,
+     * description, language and slides-link.
+     *
+     * @param $request
+     * @param $db
+     *
+     * @throws Exception
+     */
+    public function editTalkAction($request, $db)
+    {
+        $errors = array();
+        if(!isset($request->user_id)) {
+            throw new Exception("You must be logged in to edit data", 403);
+        }
+
+        $talk_id = $this->getItemId($request);
+
+        $talk_mapper = new TalkMapper($db, $request);
+
+        $existing_talk = $talk_mapper->getTalkById($talk_id, true);
+        if (! $existing_talk || $existing_talk['meta']['count'] < 1) {
+            throw new Exception('The talk could not be found', 404);
+        }
+        $existing_talk = $existing_talk['talks'][0];
+
+        $isAdmin   = $talk_mapper->thisUserHasAdminOn($talk_id);
+        //$isSpeaker = $talk_mapper->thisUserIsSpeakerOn($talk_id);
+        //if (! $isAdmin  && ! $isSpeaker) {
+        if (! $isAdmin) {
+            throw new Exception('You are not entitled to edit this entry', 403);
+        }
+
+        $eUri = explode('/', $existing_talk['event_uri']);
+        $existing_talk['event_id'] = $eUri[(count($eUri)-1)];
+
+        $talk = array(
+            'talk_id' => $talk_id,
+            'event_id' => $existing_talk['event_id']
+        );
+
+        $talk['talk_title'] = filter_var(
+            $request->getParameter("talk_title"),
+            FILTER_SANITIZE_STRING
+        );
+        if(empty($talk['talk_title'])) {
+            $errors[] = "'title' is a required field";
```

Comment:
The error messages here are repeated and are also inconsistent with the create method